xdsl/dialects/x86/ops.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Sequence, Set
+from typing import Generic, TypeAlias, TypeVar
+
+from typing_extensions import Self
+
+from xdsl.dialects.builtin import (
+    AnyIntegerAttr,
+    StringAttr,
+)
+from xdsl.ir import (
+    Attribute,
+    Operation,
+    SSAValue,
+)
+from xdsl.irdl import (
+    IRDLOperation,
+    irdl_op_definition,
+    operand_def,
+    opt_attr_def,
+    result_def,
+)
+from xdsl.parser import Parser, UnresolvedOperand
+from xdsl.printer import Printer
+
+from .register import GeneralRegisterType, X86RegisterType
+
+R1InvT = TypeVar("R1InvT", bound=X86RegisterType)
+R2InvT = TypeVar("R2InvT", bound=X86RegisterType)
+
+
+class X86Op(Operation, ABC):
+    """
+    Base class for operations that can be a part of x86 assembly printing.
+    """
+
+    @abstractmethod
+    def assembly_line(self) -> str | None:
+        raise NotImplementedError()
```

Comment:
We don't need this functionality yet in the base class, might as well add it later